File: shoestorefe/lib/domain/entities/promotion_product.dart
class PromotionProduct {
  final int productId;
  final String productName;
  final String? sku;
  final double? salePrice;
  final double discountPercent;

  PromotionProduct({
    required this.productId,
    required this.productName,
    this.sku,
    this.salePrice,
    required this.discountPercent,
  });
}

File: ShoeStore/ShoeStore.Domain/Entities/PromotionProduct.cs
namespace ShoeStore.Domain.Entities
{
    public class PromotionProduct
    {
        public long Id { get; set; }
        public int PromotionId { get; set; }
        public int ProductId { get; set; }
        public decimal DiscountPercent { get; set; }

        public Promotion Promotion { get; set; } = null!;
        public Product Product { get; set; } = null!;
    }
}
